### PR TITLE
Insight driver to kotlin, replace #529

### PR DESCRIPTION
**Pull request by Philoul**
I started to convert java files to kotlin
- I'll go as far as I can but I'll probably need help to finish convertion.... (I found some tricky files and mechanisms not very easy to manage during the work)
- Sorry for the important number of commits (I make a lot of back and forth in my work to always be able to build and test in parallel...)
- Pairing works (but I finally kept several files in java)

I just share work in progress, feel free to comment (but not ready for merging)

Current Status
- [ ] java to kotlin convertion (99% done 245 files converted / 246)
- [x] null management (replace !! in code)
- [x] code cleaning (getter, when, injection, import...)
- [x] full tests
- [x] Update with latest dev

### CODE REVIEW

**Comment by MilosKozak** [comment]
how much is it tested?

**Comment by Philoul** [comment]
I tested a lot on my dev config during the migration of each file.
All features (pairing, profile update, tbr, bolus, status, history...) worked correctly during my tests, but I was alone (note, I never tested in vivo and last run in vitro for tests was about 6 months ago).

One file is remaining in java `utils/ConnectionEstablisher.java` (I couldn't fix pairing issue with kotlin version of this file, see `ConnectionEstablisherXX.kt`).

I didn't test recently the latest version: each time I saw modifications in dev branch that generate conflicts, I updated the branch to include associated modification in kotlin files and fix merge conflicts.

I think a code review would be fine (I do my best for null management in kotlin, but it's probably not perfect...), and I can restart again to test all features with latest dev...

**Comment by MilosKozak** [comment]
maybe we could rename kt version to InsightV2 and keep old version still there until this one get more tested

**Comment by Philoul** [comment]
I started to test again latest update with my dev phone (Android 11) and my spare pump (firmware v3).
Pairing, Bolus, TBR, read history (included bolus or TBR done from pump, reservoir and site change)...
Everything works fine.
I propose to ask on discord if other users could test it with other smartphones (Android 12 for example...)

**Comment by Philoul** [comment]
Results from other users in Discord confirms that current Driver in kt seems to work like previous one in java.
Concerning Pairing process, I made 20 successive trials, 19 pairing was OK and I got only one "communication failed" durring pairing (and it worked immediatly after). So I think it's OK. I sometimes also get communication error during pairing with java driver.

A lot of `!!` remains into `LocalInsightPlugin.kt` file, but most of them are into a `try ... catch`, so I don't know if it's better remove them...

**Comment by MilosKozak** [comment]
!! should be rare, the code needs some polishing
for example this:
```
if (connectionService == null .......
 !connectionService!!.hasRequestedConnection(this)) return false
```
is not thread safe. correct is `if (connectionService?.hasRequestedConnection(this) == false) return false`


if is not necessary: `if (thread != null) thread?.interrupt()` 
check other plugins how to do it elegant way: `{ throwable: Throwable? -> fabricPrivacy.logException(throwable!!) }`

etc. just go through diff ...

**Comment by Philoul** [comment]
I removed all !! from LocalInsightPlugin file with one of my latest commit (I didn't remove them first because most of them was into try catch...)
-> I kept all the try catch (but I don't know if there are still all required with no !! into the LocalInsightDriver file...)

File: insight/src/test/java/info/nightscout/androidaps/TestBase.kt [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+open class TestBase {
+
+    val aapsLogger = AAPSLoggerTest()
+    val aapsSchedulers: AapsSchedulers = TestAapsSchedulers()
+
+    // Add a JUnit rule that will setup the @Mock annotated vars and log.
+    // Another possibility would be to add `MockitoAnnotations.initMocks(this) to the setup method.
+    @get:Rule
+    val mockitoRule: MockitoRule = MockitoJUnit.rule()
```
Comment (MilosKozak) [inline insight/src/test/java/info/nightscout/androidaps/TestBase.kt:21]: This has to be changed for junit5. See other modules

Comment (Philoul) [reply insight/src/test/java/info/nightscout/androidaps/TestBase.kt]: Done

**Comment by Philoul** [comment]
@MilosKozak, I need help to fix this crash: https://console.firebase.google.com/u/0/project/androidaps-c34f8/crashlytics/app/android:info.nightscout.androidaps/issues/7845221cec2c629ad39df892bdb1fbba?time=last-thirty-days&versions=3.1.0.3-tests_U200-dev-e%20(1500)&sessionEventKey=63A9700801C300014F0C666E39D8D66C_1759839901097333841

The App crash when I receive an alert from insight pump (ex Resevoir low, ...) and I click on a button to mute or confirm.

=> I think it's somewhere in `InsightAlertActivity.kt` https://github.com/nightscout/AndroidAPS/pull/2098/files#diff-5bfffebeef0fa7cb850ae5fca73c692f989ed660fb0e56353250ea588420592a

Thanks

**Comment by MilosKozak** [comment]
as first I'd move onClick from XML to kotlin class

**Comment by Philoul** [comment]
> as first I'd move onClick from XML to kotlin class

Thanks, it fixed the crash!

**Comment by Philoul** [comment]
I still have to rework fragment (one crash can sometimes occurs when I exit plugin fragment during a refresh)...

**Pull request closed without merging**